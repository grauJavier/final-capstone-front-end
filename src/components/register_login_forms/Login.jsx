import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import sendLogin from '../../redux/user/actions/sendLogin';
import EmailInput from './inputs/EmailInput';
import PasswordInput from './inputs/PasswordInput';
import logoFull from '../../assets/logo-full.png';

const Login = () => {
  const { register, handleSubmit, formState: { errors } } = useForm();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleClick = () => {
    navigate("/register");
  };

  const onSubmit = (data) => {
    dispatch(sendLogin(data));
  };

  return (
    <div className='bg-lime-img bg-cover bg-center min-h-screen flex flex-col items-center'>
      <img src={logoFull} className="logo-image" />
      <h2 className="secondary-font text-3xl font-bold text-zinc-50">LOGIN</h2>
      <form className="forms-layout mt-12" onSubmit={handleSubmit(onSubmit)}>
        <EmailInput register={register} errors={errors} />
        <PasswordInput register={register} errors={errors} />
        <input type="submit" className='forms-submit' value="Sign in" />
      </form>
      <p className="text-center text-white primary-font mt-auto mb-4">Don&apos;t have an account? <a className="underline cursor-pointer text-yellow-300" onClick={handleClick}>Sign up</a></p>
    </div>
  );
};

export default Login;
