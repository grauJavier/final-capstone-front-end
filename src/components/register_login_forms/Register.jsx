import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useEffect } from 'react';
import sendRegistration from '../../redux/user/actions/sendRegistration';
import UsernameInput from './inputs/UsernameInput';
import EmailInput from './inputs/EmailInput';
import PasswordInput from './inputs/PasswordInput';
import logoFull from '../../assets/logo-full.png';

const Register = () => {
  const { register, handleSubmit, formState: { errors } } = useForm();
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const data = useSelector((state) => state.user.data);

  useEffect(() => {
    if (data){
      navigate("/login");
    }
  }, [data, navigate]);

  const handleClick = () => {
    navigate("/login");
  };

  const onSubmit = (data) => {
    dispatch(sendRegistration(data));
  };

  return (
    <div className='bg-lime-img bg-cover min-h-screen flex flex-col items-center'>
      <img src={logoFull} className="logo-image" />
      <h2 className="secondary-font text-3xl font-bold text-zinc-50">REGISTER</h2>
      <form className="forms-layout" onSubmit={handleSubmit(onSubmit)}>
        <UsernameInput register={register} errors={errors} />
        <EmailInput register={register} errors={errors} />
        <PasswordInput register={register} errors={errors} />
        <input type="submit" className='forms-submit' value="Sign up" />
      </form>
      <p className="text-center text-white primary-font mt-auto mb-4">Already have an account? <a className="underline cursor-pointer text-yellow-300" onClick={handleClick}>Sign in</a></p>
    </div>
  );
};

export default Register;
