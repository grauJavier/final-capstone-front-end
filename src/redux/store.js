import { configureStore, combineReducers } from '@reduxjs/toolkit';
import userReducer from './user/userSlice';
import { persistStore, persistReducer } from 'redux-persist';
import sessionStorage from 'redux-persist/lib/storage/session'; // defaults to localStorage for web
import { createTransform } from 'redux-persist';
import placesSlice from "./places/placesSlice";
import reservationReducer from './reservation_form/reservationSlice.js';
import myReservationsReducer from './my_reservations/myReservationsSlice.js';
import noticeAlertReducer from './notice_alert/noticeAlertSlice.js';

// Create a personalized transform.
const userTransform = createTransform(
  (inboundState) => {
    return {
      currentUser: inboundState.currentUser,
      token: inboundState.token
    };
  },

  (outboundState) => {
    return {
      ...outboundState
    };
  },
  // Apply this transform only to the user state.
  { whitelist: ['user'] }
);

const persistConfig = {
  key: 'root',
  storage: sessionStorage,
  whitelist: ['user'],
  transforms: [userTransform]
};

const reducer = combineReducers({
  noticeAlert: noticeAlertReducer,
  user: userReducer,
  placesSlice,
  reservation: reservationReducer,
  myReservations: myReservationsReducer,
});

const persistedReducer = persistReducer(persistConfig, reducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false
    }),
});

const persistor = persistStore(store);

export { store, persistor };
