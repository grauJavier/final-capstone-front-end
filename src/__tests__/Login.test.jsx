import '@testing-library/jest-dom'; // For better DOM assertions
import { beforeEach, expect, describe, test } from 'vitest';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Login from '../components/register_login_forms/Login';
import Register from '../components/register_login_forms/Register';
import { store } from '../redux/store';

describe('login form', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Login />} />
            <Route path="/register" element={<Register />} />
          </Routes>
        </BrowserRouter>
      </Provider>
    );
  });

  test('should display the correct elements', async () => {
    const login = await screen.findByText('LOGIN');
    const emailInput = screen.getByPlaceholderText('Email');
    const passwordInput = screen.getByPlaceholderText('Password');
    const button = screen.getByRole('button', { name: 'Sign in' });

    expect(login).toBeInTheDocument();
    expect(emailInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });

  test('should render correctly', async () => {
    await waitFor(() => expect(Login).toMatchSnapshot());
  });

  test('should redirect to /register page when "Sign up" link is clicked', async () => {
    const signUpLink = screen.getByText('Sign up');

    fireEvent.click(signUpLink);

    const registerPage = await screen.findByText('REGISTER'); 

    expect(registerPage).toBeInTheDocument();
  });
});
