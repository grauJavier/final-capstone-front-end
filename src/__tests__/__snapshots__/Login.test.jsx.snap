// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`login form > should render correctly 1`] = `[Function]`;
